purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1485,20 +1475,22 @@ class Purchases internal constructor(
             if (!appConfig.finishTransactions) {
                 log(LogIntent.WARNING, PurchaseStrings.PURCHASE_FINISH_TRANSACTION_FALSE)
             }
-            if (!state.purchaseCallbacksByProductId.containsKey(storeProduct.productId)) {
-                val mapOfProductIdToListener = mapOf(storeProduct.productId to listener)
+            if (!state.purchaseCallbacksByProductId.containsKey(purchasingData.productId)) {
+                val mapOfProductIdToListener = mapOf(purchasingData.productId to listener)
                 state = state.copy(
                     purchaseCallbacksByProductId = state.purchaseCallbacksByProductId + mapOfProductIdToListener
                 )
                 userPurchasing = identityManager.currentAppUserID
+            } else {
+                log(LogIntent.WARNING, PurchaseStrings.PURCHASE_FINISH_TRANSACTION_FALSE)
             }
         }
+        print("🍕🍕🍕🍕")
```

Comment:
production 🍕 ?

Reply:
personally, i love it.

Reply:
My terrible testing process has been exposed 😇